File: src/main/java/com/wallet/app/service/ContactService.java
package com.wallet.app.service;

import java.util.List;
import java.util.UUID;

import com.wallet.app.dto.ContactResponse;
import com.wallet.app.dto.ContactUpsertRequest;

public interface ContactService {

    ContactResponse addContact(String username, ContactUpsertRequest request);

    ContactResponse updateContact(String username, UUID contactId, ContactUpsertRequest request);

    void deleteContact(String username, UUID contactId);

    List<ContactResponse> getContacts(String username, String query);
}

File: frontend/src/services/contact.service.ts
import type { ContactResponse, ContactUpsertRequest, MessageResponse } from '../types/api'
import { apiClient } from './apiClient'

export const contactService = {
  async getContacts(query?: string): Promise<ContactResponse[]> {
    const response = await apiClient.get<ContactResponse[]>('/contacts', {
      params: query && query.trim() ? { query: query.trim() } : undefined,
    })
    return response.data
  },

  async addContact(payload: ContactUpsertRequest): Promise<ContactResponse> {
    const response = await apiClient.post<ContactResponse>('/contacts', payload)
    return response.data
  },

  async updateContact(contactId: string, payload: ContactUpsertRequest): Promise<ContactResponse> {
    const response = await apiClient.put<ContactResponse>(`/contacts/${contactId}`, payload)
    return response.data
  },

  async deleteContact(contactId: string): Promise<MessageResponse> {
    const response = await apiClient.delete<MessageResponse>(`/contacts/${contactId}`)
    return response.data
  },
}
